### PR TITLE
Split API calls into a separate, independent API Client

### PR DESCRIPTION
**Pull request by drf**
This PR creates a new module structure, _client_, which contains a self-sufficient API Client for Astarte APIs, built according to Go best practices.

Currently, all API calls referring to existing commands have been implemented (plus interface deletion in Realm Management).

This PR also contains the full porting of all current commands to the new API client.

### CODE REVIEW

File: cmd/realm/interfaces.go [hunk 1]
```diff
@@ -235,46 +161,29 @@ func interfacesInstallF(command *cobra.Command, args []string) error {
...
+	interfaceName := args[0]
+	interfaceMajor := 0
 
-		errJson, _ := json.MarshalIndent(&errorBody, "", "  ")
-		fmt.Println(string(errJson))
+	err := astarteAPIClient.RealmManagement.DeleteInterface(realm, interfaceName, interfaceMajor, realmManagementJwt)
+	if err != nil {
+		fmt.Println(err)
 		os.Exit(1)
+		return nil
```
Comment (rbino) [inline cmd/realm/interfaces.go]: Is this even reachable? You exit from the program the line above

File: client/astarte.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+		Errors map[string]interface{} `json:"errors"`
+	}
+
+	err := json.NewDecoder(responseBody).Decode(&errorBody)
+	if err != nil {
+		return err
+	}
+
+	errJSON, _ := json.MarshalIndent(&errorBody, "", "  ")
+	return fmt.Errorf("%s", errJSON)
```
[the remark this answers is not in the document]
Comment (drf) [reply client/astarte.go]: Nope - errJSON is []byte, so it needs formatting.

File: cmd/realm/interfaces.go (hunk 1 above)
Comment (drf) [reply cmd/realm/interfaces.go]: tbh that's a very good question and I had the same doubt

Comment (drf) [reply cmd/realm/interfaces.go]: From Exit docs:
The program terminates immediately; deferred functions are not run.

So yeah, they're useless.

**Pull request merged**